config/fcos/v1_5_exp/translate.go

```diff
@@ -362,3 +362,28 @@ func buildGrubConfig(gb Grub) string {
 	superUserCmd := fmt.Sprintf("set superusers=\"%s\"\n", strings.Join(allUsers, " "))
 	return "# Generated by Butane\n\n" + superUserCmd + strings.Join(cmds, "\n") + "\n"
 }
+
+func arePartitionEqual(p1, p2 types.Partition) bool {
+    return p1.Label == p2.Label && p1.Number == p2.Number
+}
+
+
+func checkPartitionTable(disk string, wipeTable bool, partitions []types.Partition) error {
+   // if wipetable is true do nothing
+	if wipeTable {
+        return nil
+    }
+   // check partitions labels and numbers and see if there are duplicates
+
+   // problably a faster way to do this... 
+    for i, p1 := range partitions {
+        for j := i + 1; j < len(partitions); j++ {
+            p2 := partitions[j]
+            if arePartitionEqual(p1, p2) {
+                return fmt.Errorf("Duplicate partition with label and number on disk. This may cause issues during reprovisioning. Specify the root disk as /dev/disk/by-id/coreos-boot-disk to avoid this warning.", p1.Label, p1.Number, disk)
```

Comment:
All validation errors need to be defined in `errors.go` and referenced by name, so that users of the Go API can programmatically check for the error without hardcoding a string comparison.

Note that the code isn't actually checking for `/dev/disk/by-id/coreos-boot-disk` anywhere.

`Specify the root disk as ...` is technically correct, but the phrasing could mislead the user into thinking that they should specify _this_ disk as `coreos-boot-disk`, when in fact we don't know whether this disk is the boot disk or not.  Let's try to refine the phrasing a bit.

Reply:
I have the new error message committed in config/common/errors.go now!